### PR TITLE
SW-2508 Clean up checkbox border radius styles

### PR DESCRIPTION
**Pull request by karthikbtf**
- we were unable to use a border radius before using default mui checkbox
- using additional enhancements to enable this
- see Chudo's findings in the jira https://terraformation.atlassian.net/browse/SW-2508
- the focus outline is still a question on how to do
- feels like maybe the checkbox looks too big now
- test in vercel

Table - Selectable ⋅ Storybook 2022-12-22 14-07-31

### CODE REVIEW

**Comment by karthikbtf** [comment]
@hannah-cai let me know if you have any feedback.

File: src/components/Checkbox.tsx [hunk 1]
```diff
@@ -1,27 +1,50 @@
...
+      margin: '2px auto auto',
+    }
   },
-  '&:focus > .MuiSvgIcon-root': {
-    border: `2px solid ${theme.palette.TwClrBgSelected}`,
+  '& > .selected': {
+    border: '2px solid transparent',
+    backgroundColor: theme.palette.TwClrBgSelected,
+    '& > svg' : {
+      fill: 'white',
```
Comment (constanzauanini) [inline src/components/Checkbox.tsx:41]: Do we have a token for white?

File: src/components/table/index.tsx [hunk 2]
```diff
@@ -321,6 +321,7 @@ export default function EnhancedTable<T>({
                                 color='primary'
                                 checked={isItemSelected}
                                 onClick={(e) => (!isClickable || !isClickable(row as T) ? handleClick(e, row as T) : null)}
+                                {...CheckboxIcons}
```
Comment (constanzauanini) [inline src/components/table/index.tsx:324]: Could we replace the MUI checkbox with the web-component checkbox here so we don't need to pass the icons?

File: src/components/Checkbox.tsx (hunk 1 above)
Comment (karthikbtf) [reply src/components/Checkbox.tsx:41]: oops, will fix

File: src/components/table/index.tsx (hunk 2 above)
Comment (karthikbtf) [reply src/components/table/index.tsx:324]: This was not working well for some reason. I will try again.

File: src/components/Checkbox.tsx (hunk 1 above)
Comment (karthikbtf) [reply src/components/Checkbox.tsx]: fixed to use theme styles

File: src/components/table/index.tsx (hunk 2 above)
Comment (karthikbtf) [reply src/components/table/index.tsx:324]: Our checkbox is a labeled checkbox.. which has different semantics and event handling.
For now I'm going to keep them this way. We can do a refactor later. It cannot be reused the way it is now.

Comment (constanzauanini) [reply src/components/table/index.tsx:324]: Thanks for trying!

**Review by constanzauanini — APPROVED** [review]

**Comment by hannah-cai** [comment]
thank you for the updates! the checkboxes look a lot more consistent with everything else now.
a small detail I noticed: the border cuts off the checkmark on the sides.
image
I would also prefer the checkbox to be smaller, but I know that Chudo was pushing for larger checkboxes for bigger click targets. that's something we can still explore though.
I'll leave @chudo76 to be the decider on what we do for focus states, since I'm not as familiar with them.

@karthikbtf a separate issue I noticed is that table rows have a hover state when they're selected and you hover over them. also, clicking on the row selects/de-selects the checkbox; the intended behavior was that you'd need to click on the checkbox itself. these may already be covered in Rachel's table updates, but I wanted to bring them up in case they weren't

**Comment by karthikbtf** [comment]
> thank you for the updates! the checkboxes look a lot more consistent with everything else now. a small detail I noticed: the border cuts off the checkmark on the sides. image I would also prefer the checkbox to be smaller, but I know that Chudo was pushing for larger checkboxes for bigger click targets. that's something we can still explore though. I'll leave @chudo76 to be the decider on what we do for focus states, since I'm not as familiar with them.
> 
> @karthikbtf a separate issue I noticed is that table rows have a hover state when they're selected and you hover over them. also, clicking on the row selects/de-selects the checkbox; the intended behavior was that you'd need to click on the checkbox itself. these may already be covered in Rachel's table updates, but I wanted to bring them up in case they weren't

Thanks @hannah-cai ! I will take a look.

**Comment by chudo76** [comment]
> > thank you for the updates! the checkboxes look a lot more consistent with everything else now. a small detail I noticed: the border cuts off the checkmark on the sides. image I would also prefer the checkbox to be smaller, but I know that Chudo was pushing for larger checkboxes for bigger click targets. that's something we can still explore though. I'll leave @chudo76 to be the decider on what we do for focus states, since I'm not as familiar with them.
> > @karthikbtf a separate issue I noticed is that table rows have a hover state when they're selected and you hover over them. also, clicking on the row selects/de-selects the checkbox; the intended behavior was that you'd need to click on the checkbox itself. these may already be covered in Rachel's table updates, but I wanted to bring them up in case they weren't
> 
> Thanks @hannah-cai ! I will take a look.

Looks like adding `overflow: visible;` to the `svg`'s CSS fixes the clipping issue.

**Comment by sgrimm** [comment]
Closing this since we're using other checkboxes now.

**Pull request closed without merging**